src/common/mpi.jl

```diff
@@ -0,0 +1,49 @@
+# Convenience functions for working with MPI
+using MPI
+
+"""
+Initialize MPI. Must be called before doing any non-trivial MPI work
+(even in the single-process case). Unlike the MPI.Init() function,
+this can be called multiple times.
+"""
+function mpi_ensure_initialized()
+    # TODO look more closely at interaction between MPI and threads
+    MPI.Initialized() || MPI.Init_thread(MPI.ThreadLevel(3))
+end
+
+"""
+Number of processors used in MPI. Can be called without ensuring initialization.
+"""
+mpi_nprocs() = mpi_nprocs(MPI.COMM_WORLD)
+mpi_nprocs(comm) = (mpi_ensure_initialized(); MPI.Comm_size(comm))
+mpi_master() = (mpi_ensure_initialized(); MPI.Comm_rank(MPI.COMM_WORLD) == 0)
```

Comment:
I would go with the Julia standard and call this `is_mpi_master()`

Reply:
I like prefixing everything with `mpi`, makes it easier on the eye to find out which parts are MPI commands. I'd go with `mpi_is_master` but both this and `is_mpi_master` makes it sound like we're asking whether MPI is master, which is weird

Reply:
`mpi_on_master` or `on_mpi_master`?

Reply:
is that really better than mpi_master? It's clear what this does

Reply:
Ok fair enough, that's true.